precompiles/authorization/events.go

```diff
@@ -36,36 +46,34 @@ type EventRevocation struct {
 	TypeUrls []string
 }
 
-// EmitAllowanceChangeEvent creates a new allowance change event emitted on IncreaseAllowance
-// and DecreaseAllowance transactions.
-func EmitAllowanceChangeEvent(args cmn.EmitEventArgs) error {
-	// check if the provided Event is correct type
-	allowanceChangeEvent, ok := args.EventData.(EventAllowanceChange)
+// EmitRevocationEvent creates a new authorization event emitted on a Revoke transaction.
+func EmitRevocationEvent(args cmn.EmitEventArgs) error {
+	// Prepare the event topics
+	revocationEvent, ok := args.EventData.(EventRevocation)
 	if !ok {
-		return fmt.Errorf("invalid Event type, expecting EventAllowanceChange but received %T", args.EventData)
+		return fmt.Errorf("invalid Event type, expecting EventRevocation but received %T", args.EventData)
 	}
-
 	// Prepare the event topics
-	event := args.ContractEvents[EventTypeAllowanceChange]
+	event := args.ContractEvents[EventTypeRevocation]
```

Comment:
Do you think we don't need to check for the existence of the event?

```
	event, found := args.ContractEvents[EventTypeRevocation]
	if !found {
		return fmt.Errorf("event type not found: %s", EventTypeRevocation)
	}
```

Reply:
I don't think it's necessary since we are the ones creating the Interface and generating the ABI.